RecoVertex/BeamSpotProducer/plugins/OfflineToTransientBeamSpotESProducer.cc

```diff
@@ -0,0 +1,47 @@
+#include "RecoVertex/BeamSpotProducer/plugins/OfflineToTransientBeamSpotESProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Utilities/interface/do_nothing_deleter.h"
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+using namespace edm;
+
+OfflineToTransientBeamSpotESProducer::OfflineToTransientBeamSpotESProducer(const edm::ParameterSet& p) {
+  auto cc = setWhatProduced(this);
+
+  transientBS_ = new BeamSpotObjects;
+  //theOfflineBS_ = new BeamSpotObjects;
+  bsOfflineToken_ = cc.consumesFrom<BeamSpotObjects, BeamSpotObjectsRcd>();
+}
+
+OfflineToTransientBeamSpotESProducer::~OfflineToTransientBeamSpotESProducer() {
+  delete transientBS_;
```

Comment:
shall I remove the delete then? When running I didn't notice any error or warning messages, though.

Reply:
so matti, I couldnt follow this specific double delete. Is this because transientBS_ is returned as a shared pointer by the producer? But in this case, due to the deletation function, the shared pointer wont delete the object.  Or something else?

Reply:
The current version of the code is ok (no `delete`) so further discussion may be moot, but earlier in the `produce()` there was an assignment
```cpp
transientBS_ = &rec.get(bsOfflineToken_);
```
and a later `delete transientBS_` would delete an object returned by `rec.get()`, which is owned by the EventSetup system.